clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -144,21 +148,11 @@ export class CopybookDownloadService implements vscode.Disposable {
             }
         } catch (e) {
             let errorMessage = e.toString();
-            if (e instanceof ZoweError) {
-                switch (e.type) {
-                    case Type.InvalidCredentials:
-                        errorMessage = INVALID_CREDENTIALS_ERROR_MSG.replace(PROFILE_NAME_PLACEHOLDER, element.profile);
-                        this.lockedProfile.add(element.profile);
-                        this.queue.clean();
-                        TelemetryService.registerEvent("invalidCredentials", ["copybook", "COBOL", "experiment-tag"], "Zowe credentials is not valid");
-                        break;
-                    case Type.ConnRefused:
-                        errorMessage = CONN_REFUSED_ERROR_MSG.replace(PROFILE_NAME_PLACEHOLDER, element.profile);
-                        break;
-                    case Type.NoPassword:
-                        errorMessage = NO_PASSWORD_ERROR_MSG.replace(PROFILE_NAME_PLACEHOLDER, element.profile);
-                        break;
-                }
+            if (CopybookDownloadService.isInvalidCredentialsError(e)) {
+                errorMessage = INVALID_CREDENTIALS_ERROR_MSG.replace(PROFILE_NAME_PLACEHOLDER, element.profile);
+                this.lockedProfile.add(element.profile);
```

Comment:
Isn't ZE now managing the profile? And if they don't have it out-of-the-box, shouldn't we rely on their implementation?

Reply:
Naa, they don't and I ended up locking my cred. So, I kept these.
Do you suggest to remove this?

Reply:
Now it delegates to zoweAPI